src/color.c

```diff
@@ -607,6 +612,75 @@ static int parse_gnu_ls_colors(struct colors *colors, const char *ls_colors) {
 	return ret;
 }
 
+#if __FreeBSD__
+/** Parse the FreeBSD $LSCOLORS format. */
+static int parse_bsd_ls_colors(struct colors *colors, const char *lscolors) {
+	int ret = -1;
+	static const dchar *keys[LSCOLORS_KEYS] = {
```

Comment:
`dchar` is for dynamic strings from `dstring.h`.  Also, I think you can get rid of the `LSCOLORS_{KEYS,LEN}` macros and use `countof()` instead.

```suggestion
	static const char *keys[] = {
```